src/lib/components/drs/formComponents/Checksums.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { 
+    Typography,
+    FormControl,
+    TextField,
+    Grid,
+    FormGroup,
+    MenuItem
+} from '@material-ui/core';
+import SpaceDivider from '../../common/SpaceDivider';
+import AddItemButton from '../../common/AddItemButton';
+import RemoveItemButton from '../../common/RemoveItemButton';
+
+/*
+    Displayed for blob-type objects and can be added and removed. The Type
+    property is selected using a dropdown, however each type can only be
+    selected once. If a type is selected, that option is disabled for each of
+    the other checksum instances. A maximum of 3 checksum objects can be added
+    before the AddPropertyButton becomes disabled, since there are 3 type
+    options. The Checksum value field is updated through free-text entry.
+*/
+const Checksums = props => {
+
+    const includeChecksumTypeInSelection = (type, index) => {
```

Comment:
Currently there are some cases where Checksum types are displayed when they should be hidden. For example, if a user adds 3 Checksum instances, and populates each of them with one of the Checksum types, the first instance still shows all 3 options in the dropdown menu. However, the third instance only shows itself as an option that can be selected. Since all 3 options have already been selected, each instance should only display itself as an option. It seems that perhaps the type validation is only based on the types that have been selected above the current instance.